src/NuGet.Commands/RestoreCommand.cs

```diff
@@ -59,7 +59,7 @@ public RestoreCommand(ILogger logger)
             if (request.ExternalProjects.Any())
             {
                 externalProjectReference = new ExternalProjectReference(
-                    request.Project.FilePath,
+                    request.Project.Name,
```

Comment:
`Project.Name` == `Project.FilePath` when we're dealing with NuGet.CommandLine, however they are different (Name maps to UniqueName) when dealing with VS build integrated project.